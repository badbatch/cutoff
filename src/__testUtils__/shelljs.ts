import { jest } from '@jest/globals';

export const clearShelljsMock = (mock: jest.MockedObject<typeof import('shelljs')>) => {
  mock.echo.mockClear();
  mock.exec.mockClear();
  mock.exit.mockClear();
};

export const shelljsMock = () => ({
  default: {
    echo: jest.fn(),
    exec: jest.fn(),
    exit: jest.fn(),
  },
});
