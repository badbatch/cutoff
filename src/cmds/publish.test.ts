import shelljs from 'shelljs';

const { exec } = shelljs;

describe('publish', () => {
  it('should return the correct terminal output', () => {
    expect(exec('node --trace-warnings ./bin/cutoff.mjs publish --help', { silent: true }).stdout).toMatchSnapshot();
  });
});
