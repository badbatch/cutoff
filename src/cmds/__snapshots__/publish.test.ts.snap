// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`publish should return the correct terminal output 1`] = `
"cutoff.mjs publish

Publish packages to registry

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]
"
`;
